src/main/java/de/hhu/stups/plues/ui/controller/Reports.java

```diff
@@ -185,67 +270,242 @@ private void displayModuleAbstractUnitUnitSemesterConflicts(final Store store,
+              store.getAbstractUnitById(pair.getSecond()))).collect(Collectors.toSet())))));
+    this.impossibleCourseModuleAbstractUnitPairs.setData(
+        impossibleCourseModuleAbstractUnitPairs);
+
+    return impossibleCourseModuleAbstractUnitPairs;
   }
 
-  private void displayImpossibleCourseModuleAbstractUnits(final Store store,
-      final ReportData reportData) {
-    impossibleCourseModuleAbstractUnits.setData(reportData.getImpossibleCourseModuleAbstractUnits()
+  private Map<Course, Map<Module, Set<AbstractUnit>>>
+      displayImpossibleCourseModuleAbstractUnits(final Store store, final ReportData reportData) {
+    Map<Course, Map<Module, Set<AbstractUnit>>> impossibleCourseModuleAbstractUnits =
+        reportData.getImpossibleCourseModuleAbstractUnits()
         .entrySet().stream().collect(Collectors.toMap(
           entry -> store.getCourseByKey(entry.getKey()),
           entry -> entry.getValue().entrySet().stream().collect(Collectors.toMap(
             innerEntry -> store.getModuleById(innerEntry.getKey()),
             innerEntry -> innerEntry.getValue().stream().map(
-                store::getAbstractUnitById).collect(Collectors.toSet()))))));
+            store::getAbstractUnitById).collect(Collectors.toSet())))));
+    this.impossibleCourseModuleAbstractUnits.setData(impossibleCourseModuleAbstractUnits);
+
+    return impossibleCourseModuleAbstractUnits;
   }
 
-  private void displayRedundantUnitGroups(final Store store, final ReportData reportData) {
-    redundantUnitGroups.setData(reportData.getRedundantUnitGroups().keySet().stream()
-        .map(store::getUnitById).collect(Collectors.toSet()));
+  private Set<Unit> displayRedundantUnitGroups(final Store store, final ReportData reportData) {
+    Set<Unit> redundantUnitGroups = reportData.getRedundantUnitGroups().keySet().stream()
+        .map(store::getUnitById).collect(Collectors.toSet());
+    this.redundantUnitGroups.setData(redundantUnitGroups);
+
+    return redundantUnitGroups;
   }
 
-  private void displayQuasiMandatoryModuleAbstractUnits(final Store store,
-      final ReportData reportData) {
-    quasiMandatoryModuleAbstractUnits.setData(reportData.getQuasiMandatoryModuleAbstractUnits()
+  private Map<Module, Set<AbstractUnit>>
+      displayQuasiMandatoryModuleAbstractUnits(final Store store, final ReportData reportData) {
+    Map<Module, Set<AbstractUnit>> quasiMandatoryModuleAbstractUnits =
+        reportData.getQuasiMandatoryModuleAbstractUnits()
         .entrySet().stream().collect(Collectors.toMap(
           entry -> store.getModuleById(entry.getKey()),
           entry -> entry.getValue().stream().map(
-              store::getAbstractUnitById).collect(Collectors.toSet()))));
+          store::getAbstractUnitById).collect(Collectors.toSet())));
+    this.quasiMandatoryModuleAbstractUnits.setData(quasiMandatoryModuleAbstractUnits);
+
+    return quasiMandatoryModuleAbstractUnits;
```

Comment:
split into a method that computes the data (e.g. in RealReportData) and one step that displays it.